### PR TITLE
Fix(docker) Modules SQL do not get updated or populated for docker set ups

### LINKED ISSUE #18965: Docker WSL: Modules won't import SQL data
### Current Behaviour

I can't seem to figure this one out. I installed AzerothCore via Docker on Windows 10 using WSL. Things seemed to run fine, docker images are running, I can connect with a client, create characters, run around, use in-game commands and everything.

Then I tried to install the basic [skeleton module](https://github.com/azerothcore/skeleton-module) and after a few tweaks to the file names I got it to compile. But for some reason it just won't import the sql data (in this case the hello world acore string). I then tried different mods, and all seem to be suffering from this. No imports to the database. Therefore no mod seems to be working.

Here you can see the [ac-worldserver logs](https://gist.github.com/gruselgg/12cfb9c4cf63aa33888cd2c06d58d106).

Errors I discovered so far:
- When logging in with a character: In the ac-worldserver console I get `Acore string entry 35410 not found in DB.` (ac-worldserver logs last line)
- When logging in with a character: In-game I get the message `<error>` in the chat
- When starting the docker containers i get in the ac-worldserver `Can't set process priority class, error: Permission denied` right at the start (ac-worldserver logs line 24)

I'm not sure the last error is relevant to this problem, but it seems off.

### Expected Blizzlike Behaviour

Modules should work.

### Source

_No response_

### Steps to reproduce the problem

1. Install AzerothCore via Docker and WSL on Windows 10.
2. Add a module (like the basic skeleton module) and compile again.
3. Log in with a character and see errors in ac-worldserver console and in-game.

### Extra Notes

Things I tried:
- Applying fixes from #92
- Changing file permissions
- Using/not using `sudo` and checking user groups
- Reinstalling

### AC rev. hash/commit

AzerothCore rev. 46aed4dec768+ 2024-05-28 06:44:38 +0000 (master branch) (Unix, RelWithDebInfo, Static)

### Operating system

Windows 10

### Custom changes or Modules

- [skeleton module](https://github.com/azerothcore/skeleton-module)

### PR DESCRIPTION
**Pull request by ben-of-codecraft**
## Changes Proposed:
This PR proposes changes to:
-  [ ] Core (units, players, creatures, game systems).
-  [ ] Scripts (bosses, spell scripts, creature scripts).
-  [ ] Database (SAI, creatures, etc).
-  [X] Apps(Docker)

## Issues Addressed:
Related issues:
* https://github.com/azerothcore/azerothcore-wotlk/issues/18965
* https://github.com/azerothcore/azerothcore-wotlk/pull/16155

Support dicussion: 
https://discord.com/channels/217589275766685707/1280317898140680307
- Closes #18965

## SOURCE:
The changes have been validated through:
- [ ] Live research (checked on live servers, e.g Classic WotLK, Retail, etc.)
- [ ] Sniffs (remember to share them with the open source community!)
- [ ] Video evidence, knowledge databases or other public sources (e.g forums, Wowhead, etc.)
- [X] The changes promoted by this pull request come partially or entirely from another project (cherry-pick). 

## Tests Performed:
This PR has been:
- [X] Tested in-game by the author.
- [ ] Tested in-game by other community members/someone else other than the author/has been live on production servers.
- [ ] This pull request requires further testing and may have edge cases to be tested.

## How to Test the Changes:
- [ ] This pull request can be tested by following the reproduction steps provided in the linked issue
- [ ] This pull request requires further testing. Provide steps to test your changes. If it requires any specific setup e.g multiple players please specify it as well.

1. Install using docker instructions
2. Add a module that has sql in it mod-guildhouse as an example
3. Enabled confgiruation options in dbimport.conf.    Updates.AllowedModules to either "all" or the name of the module you are testing with. 
4. Validate in dbimport container start logging stating the module was loaded and the sql was run
5. Validate the tables or queries ran on the database

 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

### CODE REVIEW

**Review by Helias — APPROVED** [review]

File: src/tools/dbimport/dbimport.conf.dist [hunk 1]
```diff
@@ -159,6 +159,17 @@ CharacterDatabase.SynchThreads = 1
...
+#
+#   Updates.AllowedModules
+#       Description: A list of modules that are allowed to be updated automatically by the DBImport tool.
+#                    If the list is empty, no modules are allowed to automatically update. (current behavior)
+#       Default:     "" - (No modules are allowed)
+#
+#       Example:     "mod_name,mod_name2,mod_name3" (selected modules)
+#                     "all" - (All modules are allowed)
+#
+Updates.AllowedModules = ""
```
Comment (Helias) [inline src/tools/dbimport/dbimport.conf.dist:171]: I would put `"all"` as default

File: src/tools/dbimport/Main.cpp [hunk 2]
```diff
@@ -103,8 +103,15 @@ bool StartDB()
 {
     MySQL::Library_Init();
 
-    // Load databases
-    DatabaseLoader loader("dbimport");
+    // Load modules conditionally based on what modules are allowed to auto-update or none
+    std::string modules = sConfigMgr->GetOption("Updates.AllowedModules", std::string(""));
```
Comment (Helias) [inline src/tools/dbimport/Main.cpp:107]: I don't think that the casting to `std::string` is necessary here, you could just put `""`

Comment (Helias) [reply src/tools/dbimport/Main.cpp:107]: Moreover, you should set `GetOption<std::string>`

File: src/tools/dbimport/dbimport.conf.dist (hunk 1 above)
Comment (ben-of-codecraft) [reply src/tools/dbimport/dbimport.conf.dist:171]: That was my initial thought as well, in testing though I did encounter modules that break the build if they have already been applied.  I was trying to keep it as backwards compatible as possible. Maybe there is a way to figure out a 1 time initial fix for people, document it, then convert it over to all in the future?

File: src/tools/dbimport/Main.cpp (hunk 2 above)
Comment (ben-of-codecraft) [reply src/tools/dbimport/Main.cpp:107]: Thanks for feedback, I am not a C++ dev as I do full stack web dev and devops as my day job.

Will make the change and commit back.

File: src/tools/dbimport/dbimport.conf.dist (hunk 1 above)
Comment (Helias) [reply src/tools/dbimport/dbimport.conf.dist:171]: I would put "all" and then fix the related modules that break the build 
I can fix them if the build does not pass

Comment (ben-of-codecraft) [reply src/tools/dbimport/dbimport.conf.dist:171]: Ok I will do that.

**Review by Helias — APPROVED** [review]

**Comment by Helias** [comment]
I tested it locally, I think this can be merged

**Comment by Nyeriah** [comment]
Since this has been tested, reviewed and approved, I'll be merging it now. Thanks for the PR

**Comment by TheSCREWEDSoftware** [comment]
CC @ben-of-codecraft related to this PR and issue: https://github.com/azerothcore/azerothcore-wotlk/issues/21389

This PR has broken trying to build tools seperately.

**Pull request merged**